### PR TITLE
Display name

### LINKED ISSUE #44: Render user.displayName as minecraft-nickname
So like if you login as yoo-nah@transcend.vr, yoo-nah's avatar displays 'TheChair' over her head.

### PR DESCRIPTION
**Pull request by JoeyDarbyshire**
Resolves #44

### CODE REVIEW

File: server/socket.js [hunk 1]
```diff
@@ -10,11 +10,28 @@ const { getOtherUsers } = require('./utils');
...
 
+    socket.on('connectUser', (user) => {
+      socket.createdUser = true;
+      store.dispatch(createAndEmitUser(socket, user));
+    });
+
+    socket.on('sceneLoad', () => {
+      console.log('Scene loaded');
+      socket.sceneLoaded = true;
+      if (socket.createdUser === true) {
```
Comment (parky22) [inline server/socket.js]: could just change to if(socket.createdUser)

File: server/redux/reducers/user-reducer.js [hunk 2]
```diff
@@ -37,14 +37,17 @@ const removeUser = userId => {
...
-    const user = Map(createUser(userId));
-    dispatch(addUser(user));
-    socket.on('sceneLoad', () => {
-      socket.emit('createUser', user);
-    });
+    const displayName = user.displayName;
+    const newUser = Map(createUser(userId, displayName));
+    dispatch(addUser(newUser));
+    console.log('User created');
+    if (socket.sceneLoaded === true) {
```
Comment (parky22) [inline server/redux/reducers/user-reducer.js]: if(socket.sceneLoaded)

**Review by parky22 — APPROVED** [review]

File: server/redux/reducers/user-reducer.js [hunk 3]
```diff
@@ -37,14 +37,16 @@ const removeUser = userId => {
...
-    const user = Map(createUser(userId));
-    dispatch(addUser(user));
-    socket.on('sceneLoad', () => {
-      socket.emit('createUser', user);
-    });
+    const displayName = user.displayName;
+    const newUser = Map(createUser(userId, displayName));
+    dispatch(addUser(newUser));
+    console.log('User created');
+    if (socket.sceneLoaded) {
```
Comment (bushidocodes) [inline server/redux/reducers/user-reducer.js]: What happens here if the scene hasn't loaded by this point? Is this a one shot deal where the user's and controls never load if it's not loaded at this point?

Comment (JoeyDarbyshire) [reply server/redux/reducers/user-reducer.js]: Nothing.  If user is created first, then when sceneLoad is emitted, the server will send back createUser then.  Basically this line and line 28 in server/socket.js do the same thing, but solve the race condition.  It doesn't matter which happens first, the createUser emit won't fire until both happen.

File: server/redux/reducers/user-reducer.js [hunk 4]
```diff
@@ -37,14 +37,16 @@ const removeUser = userId => {
...
+      socket.emit('createUser', newUser);
```
Comment (bushidocodes) [inline server/redux/reducers/user-reducer.js]: At this point, it seems like maybe this 'createUser' event should be 'renderAvatar' or something more semantically appropriate. It's way to similar to the addUser and createUser functions.

File: server/redux/reducers/user-reducer.js [hunk 5]
```diff
@@ -37,14 +37,16 @@ const removeUser = userId => {
 
 /* --------------- THUNK ACTION CREATORS --------------- */
 
-const createAndEmitUser = socket => {
+const createAndEmitUser = (socket, user) => {
   return dispatch => {
     const userId = socket.id;
-    const user = Map(createUser(userId));
```
Comment (bushidocodes) [inline server/redux/reducers/user-reducer.js:43]: This createUser seems a bit confusing to me. Is there a reason we can't just export and use the User constructor directly to make things a bit clearer?

**Review by bushidocodes — CHANGES_REQUESTED** [review]
Code quality is excellent. I'm just finding our names to be a bit semantically confusing.

**Review by bushidocodes — APPROVED** [review]
Great work from a great American!

**Pull request merged**